gym/vector/__init__.py

```diff
@@ -0,0 +1,56 @@
+from gym.vector.async_vector_env import AsyncVectorEnv
+from gym.vector.sync_vector_env import SyncVectorEnv
+from gym.vector.vector_env import VectorEnv
+
+__all__ = ['AsyncVectorEnv', 'SyncVectorEnv', 'VectorEnv', 'make']
+
+def make(id, num_envs=1, asynchronous=True, episodic=False, **kwargs):
```

Comment:
When is episodic=True used?  I can't reset individual environments, so I'd have to reset them all at once right?

Reply:
Yes, `episodic=True` means that the individual environments do not reset automatically after finishing an episode. You get only a single episode of each environment, and you are responsible for resetting the environment yourself. This is meant to cover the frequent pattern
```python
while not done
    env.step(action)
```

Reply:
How does that cover that pattern?  Seems like I want episodic=True in that case or else I'll step past the end of some environments.

Reply:
Yes, in that case you'd like `episodic=True`. Here is an example
```python
import gym
import numpy as np

env = gym.vector.make('CubeCrash-v0', 5, episodic=True)
observations = env.reset()
dones = np.zeros(env.num_envs, dtype=np.bool_)
# This terminates after getting 1 episode per environment
while not dones.all():
    actions = env.action_space.sample()
    observations, rewards, dones, _ = env.step(actions)

env = gym.vector.make('CubeCrash-v0', 5, episodic=False)
observations = env.reset()
dones = np.zeros(env.num_envs, dtype=np.bool_)
# This doesn't terminate
while not dones.all():
    actions = env.action_space.sample()
    observations, rewards, dones, _ = env.step(actions)
```

This allows you to either
 - Get a fixed amount of episodes. Let's say you have 5 environments and you want 20 episodes overall, with `episodic=True` you can just loop 4 times through this (and you are responsible for calling `reset`):
```python
env = gym.vector.make('CubeCrash-v0', 5, episodic=True)
for _ in range(4):
    observations = env.reset()
    dones = np.zeros(env.num_envs, dtype=np.bool_)

    while not dones.all():
        actions = env.action_space.sample()
        observations, rewards, dones, _ = env.step(actions)
```
 - Or get an unbounded amount of episodes, without having to reset the environment yourself. This is the behavior you get in baselines, and is the default one here.

Reply:
I think this is a little dangerous since it's easy to confuse the past-done "undefined" observations with the good ones in this case.  What we normally do is if you want 20 episodes you do `episodic=False` and then count the number of dones (`done_count += np.sum(dones)` for instance).   Any thoughts @pzhokhov ?

Reply:
The episodic=True seems dangerous for me as well (left a separate comment in the code about that); while I can see the cases where it is useful, I think proper implementation will require env-specific details (like what is an ok observation after done, what is in the info dictionary).